src/full_energy.c

```diff
@@ -206,26 +237,179 @@ int  full_energy(int                  gridcell,
+		  moisttotal1+=moistlayer;
+		  all_vars_crop->cell[1][band].layer[lidx].moist = (all_vars_crop->cell[1][band].layer[lidx].moist*old_crop_frac+all_vars_crop->cell[0][band].layer[lidx].moist*(new_crop_frac-old_crop_frac))/new_crop_frac;
+		  all_vars_crop->cell[0][band].layer[lidx].moist = all_vars_crop->cell[0][band].layer[lidx].moist;
+		  moistlayer=all_vars_crop->cell[0][band].layer[lidx].moist*(1-new_crop_frac)+all_vars_crop->cell[1][band].layer[lidx].moist*new_crop_frac;
+		  moisttotal2+=moistlayer;
+		  //all_vars_crop->snow[1][band].swq=(all_vars_crop->snow[1][band].swq*old_crop_frac+all_vars_crop->snow[0][band].swq*(new_crop_frac-old_crop_frac))/new_crop_frac;
+		  //all_vars_crop->snow[0][band].swq=all_vars_crop->snow[0][band].swq;
+		}
+	      }
+	      else { // fallow fraction is growing
+		for(lidx=0;lidx<options.Nlayer;lidx++) {
+		  all_vars_crop->cell[0][band].layer[lidx].moist = (all_vars_crop->cell[0][band].layer[lidx].moist*(1-old_crop_frac)+all_vars_crop->cell[1][band].layer[lidx].moist*(old_crop_frac-new_crop_frac))/(1-new_crop_frac);
+		}
+	      }
+	      all_vars_crop->veg_var[1][band].Wdew *= old_crop_frac/new_crop_frac;
+	      all_vars_crop->snow[1][band].snow_canopy *= old_crop_frac/new_crop_frac;
+	    }
+	    veg_var[iveg][band].crop_frac = new_crop_frac;
+	  }
+	}
+      }
+    }
+  }
+
+  /* Assign current veg albedo and LAI (standard vic veg ) */
   if (rec >= 0) {
     // Loop over vegetated tiles
     for(iveg = 0; iveg < Nveg; iveg++){
       veg_class = veg_con[iveg].veg_class;
-      if (veg_hist[rec][iveg].vegcover[0] < MIN_VEGCOVER)
+        if (veg_hist[rec][iveg].vegcover[0] < MIN_VEGCOVER)
         veg_hist[rec][iveg].vegcover[0] = MIN_VEGCOVER;
       for ( band = 0; band < Nbands; band++ ) {
         veg_var[iveg][band].vegcover = veg_hist[rec][iveg].vegcover[0];
-        veg_var[iveg][band].albedo = veg_hist[rec][iveg].albedo[0];
+	veg_var[iveg][band].albedo = veg_hist[rec][iveg].albedo[0];
         veg_var[iveg][band].LAI = veg_hist[rec][iveg].LAI[0];
-        // Convert LAI from global to local
-        veg_var[iveg][band].LAI /= veg_var[iveg][band].vegcover;
-        veg_var[iveg][band].Wdew /= veg_var[iveg][band].vegcover;
-        veg_var[iveg][band].Wdmax = veg_var[iveg][band].LAI*LAI_WATER_FACTOR;
-        snow[iveg][band].snow_canopy /= veg_var[iveg][band].vegcover;
+ 	       veg_var[iveg][band].Wdmax = veg_var[iveg][band].LAI*LAI_WATER_FACTOR;
+        // Handle crop tiles (fallow and crop sub-tiles)
+        if (options.CROPFRAC && veg_con[iveg].crop_frac_active) {
+          for (cridx=veg_con[iveg].crop_frac_idx; cridx<veg_con[iveg].crop_frac_idx+2; cridx++) {
+            if (cridx % 2 == 0) { //fallow part
+              all_vars_crop->veg_var[cridx][band].vegcover = MIN_VEGCOVER;
+              all_vars_crop->veg_var[cridx][band].albedo = BARE_SOIL_ALBEDO;
+              all_vars_crop->veg_var[cridx][band].LAI = 0;
```

Comment:
I would suggest adding the following lines here:
all_vars_crop->veg_var[cridx][band].Wdew = 0;
all_vars_crop->veg_var[cridx][band].Wdmax = 0;
all_vars_crop->snow[cridx][band].snow_canopy = 0;